UKTrains.indigoPlugin/Contents/Server Plugin/nredarwin/webservice.py

```
@@ -1,763 +1,341 @@
+                url = f'{url}?{urllib.parse.urlencode(cleaned)}'
+
+        req = urllib.request.Request(
+            url,
+            headers={
+                'x-apikey': self._api_key,
+                'Accept': 'application/json',
+                # Apigee at api1.raildata.org.uk rejects the default
+                # `Python-urllib/x.y` agent with 403; send a plain UA.
+                'User-Agent': 'UKTrains-IndigoPlugin/2026.1',
+            },
         )
```

Ruff (0.15.12)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)